File: frontend/src/lib/types/WeatherInfo.ts
import type GeneralWeatherData from "./GeneralWeatherData.ts";
import type RainData from "./RainData.ts";
import type WindData from "./WindData.ts";

export default interface WeatherInfo {
    generalData: GeneralWeatherData,
    rainData: RainData[],
    windData: WindData
}

File: backend/app/src/main/java/io/jp/core/domain/weather/info/WeatherInfo.java
package io.jp.core.domain.weather.info;

import io.jp.core.domain.weather.general.GeneralData;
import io.jp.core.domain.weather.rain.RainData;
import io.jp.core.domain.weather.wind.WindData;
import lombok.Builder;

import java.util.List;

@Builder
public record WeatherInfo(GeneralData generalData, List<RainData> rainData, WindData windData) {
}
